### PR TITLE
Improve description of conditionals + comments in spec documentation

### PR DESCRIPTION
**Pull request by pavlinamv**
- Improve description of conditionals in spec documentation
- Add description of comments in spec documentation

### CODE REVIEW

File: doc/manual/spec [hunk 1]
```diff
@@ -214,24 +214,67 @@ can express this as
...
 \endverbatim
 
+\section comments Comments
+
+Comments in spec file have # at the start of the line.
+\verbatim
+	# this is a comment
+\endverbatim
+Macros are expanded in comments, thus if a macro is in comment use two percent
+characters before it:
```
Comment (pmatilai) [inline doc/manual/spec]: I'd make that more like:
```
Macros are expanded even in comment lines. If this is undesireable, escape the macro with an extra percent sign (%):
<...example..>
```

Now that we have such a thing, it'd be good to mention %dnl as a less quirky way of adding comments to specs.

File: doc/manual/spec [hunk 2]
```diff
@@ -214,24 +214,67 @@ can express this as
...
 \verbatim
-	%ifarch ppc64 ppc x86_64
-	...
-	%if 0%{?ver} && 0%{?ver} >= 100
-	...
-	%else
-	...
-	%endif
+	%ifarch %{power64} s390x x86_64 ia64
+	%global architectures 1
```
Comment (pmatilai) [inline doc/manual/spec]: The new example is just as meaningles as the old one. While we're at it, lets make the example actually meaningful. As in, something that actually explains why somebody would want to use it, such as this snippet from anaconda.spec:
```
%ifarch s390 s390x
BuildRequires: s390utils-devel
%endif
```

File: doc/manual/spec [hunk 3]
```diff
@@ -214,24 +214,67 @@ can express this as
...
 	%endif
 \endverbatim
 
-Conditionals are not macros, thus they can be used in spec files only.
+%ifos is used to control RPM's spec file processing according to the
+build system's operating system.
```
Comment (pmatilai) [inline doc/manual/spec]: It's not build system's operating system, but the build *target* operating system. In > 99% of the cases that's the same thing, but in theory rpm supports cross-architecture and -os builds.

File: doc/manual/spec [hunk 4]
```diff
@@ -214,24 +214,67 @@ can express this as
...
+
+%if can be used for various purposes. The test can be evaluated based on
+the existence of a macro, like:
+\verbatim
+	%if %{defined with_foo} && %{undefined with_bar}
+\endverbatim
+string comparison:
+\verbatim
+	%if %{optimize_flags} != "none"
```
Comment (pmatilai) [inline doc/manual/spec]: To set a good example, for string comparisons it's best to quote both operands.

File: doc/manual/spec [hunk 5]
```diff
@@ -214,24 +214,67 @@ can express this as
...
+\endverbatim
+or a mathematical statement:
+\verbatim
+	%if 0%{?fedora} > 10 || 0%{?rhel} > 7
+\endverbatim
+Generally, a mathematical statement allows to use logical operators
+&&, ||, !, relational operators !=, ==, <, > , <=, >= and parentheses.
```
Comment (pmatilai) [inline doc/manual/spec]: Might as well document the rest of the mathematical operators as well (+-/*)

File: doc/manual/spec [hunk 6]
```diff
@@ -214,24 +214,67 @@ can express this as
...
+of a definition like
+\verbatim
+	%define macro_with_if \
+	%if 0  \
+	%global output XXX\
+	%endif
+\endverbatim
+the expansion (including %global) occurs before evaluating of the
+conditionals. Thus after the expansion of %macro_with_if macro %output
+is defined to XXX.
```
Comment (pmatilai) [inline doc/manual/spec]: I don't think this last block belongs here, at least in this detail. Dark corners are often best left undocumented to allow changing if a saner behavior becomes feasible later. I'd condense that into something like:

%if-conditionals are not macros, and are unlikely to yield expected results if used in them.

**Comment by pavlinamv** [comment]
The comments are incorporated in the new version.
Thank you for the review.

**Comment by pmatilai** [comment]
Works for me.

**Pull request merged**